cmd/copy.go

```diff
@@ -816,180 +813,27 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 		CredentialInfo: cca.credentialInfo,
 	}
 
-	// TODO remove this copy pasted code during refactoring
 	from := cca.fromTo.From()
 	to := cca.fromTo.To()
+
 	// Strip the SAS from the source and destination whenever there is SAS exists in URL.
 	// Note: SAS could exists in source of S2S copy, even if the credential type is OAuth for destination.
-	switch from {
-	case common.ELocation.Local():
-		tmpSrc, err := filepath.Abs(cca.source)
-		if err != nil {
-			return fmt.Errorf("couldn't get absolute path of the source location %s. Failed with errror %s", cca.source, err.Error())
-		}
-
-		tmpSrc = common.ToExtendedPath(tmpSrc)
-
-		jobPartOrder.SourceRoot = cleanLocalPath(getPathBeforeFirstWildcard(tmpSrc))
-		cca.source = cleanLocalPath(tmpSrc)
-
-	case common.ELocation.Benchmark():
-		// noop
-
-	case common.ELocation.Blob():
-		fromUrl, err := url.Parse(cca.source)
-		if err != nil {
-			return fmt.Errorf("error parsing the source url %s. Failed with error %s", fromUrl.String(), err.Error())
-		}
-		blobParts := azblob.NewBlobURLParts(*fromUrl)
-		cca.sourceSAS = blobParts.SAS.Encode()
-		jobPartOrder.SourceSAS = cca.sourceSAS
-		blobParts.SAS = azblob.SASQueryParameters{}
-		bUrl := blobParts.URL()
-		cca.source = bUrl.String()
-
-		// set the clean source root
-		if strings.Contains(blobParts.ContainerName, "*") {
-			if blobParts.BlobName != "" {
-				return errors.New("cannot combine a wildcarded container name and blob path")
-			}
-
-			blobParts.ContainerName = ""
```

Comment:
What happened to this check, re can't combine wildcard container name and blob path?  Is something else already checking that elsewhere?

Reply:
And its not just the validation check that's gone AWOL ;-).  It's also the stripping of container names that contain * out of the source root, which is important too (else source root functionality will be broken and, even in the best case, won't therefore optimize the file size in plan files)